slither/solc_parsing/declarations/function.py

```diff
@@ -663,19 +663,123 @@ def _parse_try_catch(self, statement: Dict, node: NodeSolc) -> NodeSolc:
 
         if externalCall is None:
             raise ParsingError(f"Try/Catch not correctly parsed by Slither {statement}")
+
+        try_clause = statement["clauses"][0]
+        catch_clauses = statement["clauses"][1:]
+        try_params = try_clause["parameters"]["parameters"] \
+                     if "parameters" in try_clause \
+                     else []
+        new_node = node
+        if len(try_params) > 0:
+            var_identifiers = []
+            for v in try_params:
+                identifier = {
+                    "nodeType": "Identifier",
+                    "referencedDeclaration": v["id"],
+                    "src": v["src"],
+                    "name": v["name"],
+                    "typeDescriptions": {"typeString" : v["typeDescriptions"]["typeString"]}
+                }
+                var_identifiers.append(identifier)
+
+                new_statement = {
+                    "nodeType": "VariableDeclarationStatement",
+                    "src": v["src"],
+                    "declarations": [v],
+                    "initialValue": None,
+                }
+                new_node = self._parse_variable_definition(new_statement, new_node)
+            tuple_expression = {
+                "nodeType" : "TupleExpression",
+                "src": statement["src"],
+                "components": var_identifiers
+            }
+            try_expr = {
+                "nodeType": "Assignment",
+                "src": statement["src"],
+                "operator": "=",
+                "type": "tuple()",
+                "leftHandSide": tuple_expression,
+                "rightHandSide": externalCall,
+                "typeDescriptions": {"typeString": "tuple()"}
+            }
+        else:
+            try_expr = externalCall
+
         catch_scope = Scope(
-            node.underlying_node.scope.is_checked, False, node.underlying_node.scope
+            new_node.underlying_node.scope.is_checked, False, new_node.underlying_node.scope
         )
-        new_node = self._new_node(NodeType.TRY, statement["src"], catch_scope)
-        new_node.add_unparsed_expression(externalCall)
-        link_underlying_nodes(node, new_node)
-        node = new_node
+        try_catch_node = self._new_node(NodeType.TRY, statement["src"], catch_scope)
+        try_catch_node.add_unparsed_expression(try_expr)
+        link_underlying_nodes(new_node, try_catch_node)
 
-        for clause in statement.get("clauses", []):
-            self._parse_catch(clause, node)
-        return node
+        self._parse_catch(try_clause, try_catch_node, True)
```

Comment:
I realize Slither does not provide a way to add temporary variables in AST. So, a compromised CFG would be like
```
A sequence of variable declarations for try-returns variables (if necessary)
TRY node w/ tuple assignment expression and externalCall
- CATCH node for the try-clause
    - try-clause body
- CATCH node for the catch-clause
    - catch-clause body
- additional CATCH nodes (if any)
```
This CFG is a bit inaccurate since returns-variables are initialized even in the case of an exception. But, it's an acceptable compromise.